tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,306 @@
+
+    // If consent doesn't already exist in `consent` table, insert
+    $exists = $db->pselectOne(
+        "SELECT * FROM consent WHERE Name=:consentname",
+        array('consentname' => $consentName)
+    );
+    if (empty($exists)) {
+        $db->insert(
+            'consent',
+            array(
+               'Name'  => $consentName,
+               'Label' => $consentLabel,
+             )
+        );
+    }
+    // Save ConsentID inserted
+    $consentID = $db->pselectOne(
+        "SELECT ConsentID FROM consent WHERE Name=:consentName",
+        array('consentName' => $consentName)
+    );
+    // Get all data where consent status has a value in participant_status
+    // Consent status without value means the candidate does not have this specific study consent
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $dataQuery         = "SELECT CandID, " . $db->escape($consentName) . ", " . $db->escape($consentDate) . ", " . $db->escape($consentWithdrawal) . " FROM participant_status 
+                         WHERE " . $db->escape($consentName) . " IS NOT NULL OR " . $db->escape($consentName) . " <> ''";
+    $psData            = $db->pselect($dataQuery, array());
+    foreach ($psData as $entry) {
+        // Format data
+        $consentValues = [
+            'CandidateID'   => $entry['CandID'],
+            'ConsentID'     => $consentID,
+            'Status'        => $entry[$consentName],
+            'DateGiven'     => $entry[$consentDate],
+            'DateWithdrawn' => $entry[$consentWithdrawal],
+        ];
+        // Push each formatted entry to array
+        array_push($dataArray, $consentValues);
+    }
+}
+// Output list of data to be inserted into new table
+echo "\nRows to be inserted into 'candidate_consent_rel' table ..\n";
+echo "Empty dates will be inserted as NULL.\n\n";
+print_r($dataArray);
+
+// Populate candidate_consent_rel
+foreach ($dataArray as $consentValues) {
+    $db->insert('candidate_consent_rel', $consentValues);
+}
+echo "\nConsent data insert complete.\n";
+
+// Select consent history and import into new history table
+$historyFieldsQuery = "SELECT CandID, entry_staff, data_entry_date";
+
+foreach ($consentList as $consentName=>$consentLabel) {
+    $historyFieldsQuery .= ", " . $db->escape($consentName) . ", " . $db->escape($consentName) . "_date, " . $db->escape($consentName) . "_withdrawal";
+}
+$historyFieldsQuery .= " FROM consent_info_history";
+$consentHistory = $db->pselect($historyFieldsQuery, array());
+
+foreach ($consentHistory as $entry) {
+    $candID = $entry['CandID'];
+    $pscid = $db->pselectOne(
+        "SELECT PSCID FROM candidate WHERE CandID=:cid",
+        array('cid' => $candID)
+    );
+    $entryStaff = $entry['entry_staff'];
+    $entryDate = $entry['data_entry_date'];
+
+    foreach($consentList as $consentName=>$consentLabel) {
+        if(array_key_exists($consentName, $entry)) {
```

Comment:
if you dont enter this IF statement, the $db->insert is run anyways with empty parameters !

Reply:
merci merci

Reply:
sorry this comment should be on line 273